bundles/io/org.eclipse.smarthome.io.rest.core/src/main/java/org/eclipse/smarthome/io/rest/core/internal/profile/ProfileTypeResource.java

```diff
@@ -111,4 +195,13 @@ public void setProfileTypeRegistry(ProfileTypeRegistry registry) {
     public void unsetProfileTypeRegistry(ProfileTypeRegistry registry) {
         this.profileTypeRegistry = null;
     }
+
+    @Reference(cardinality = ReferenceCardinality.OPTIONAL, policy = ReferencePolicy.DYNAMIC)
+    public void setChannelTypeRegistry(ChannelTypeRegistry registry) {
```

Comment:
by default we have the DS binding methods with modifier `protected` instead of `public`. This prevents others from accidentally "destroying" an existing service. Although the `ProfileTypeResource` is not injected into other services I suggest following this approach.
In the test case you are nonetheless able to call these methods with a little trick. Instantiate the `ProfileTypeResource` like this:

```
ptr = new ProfileTypeResource() {
    {
        setChannelTypeRegistry(...);
        setProfileTypeRegistry(...);
    }
};
```

Reply:
Thanks, I didn't notice so far that we use `protected` visibility now. I have changed in in the resource. And since the test is in the same package I can also call them without a problem, so no need for a workaround there.